docs/gcp-openshift.md

```diff
@@ -0,0 +1,105 @@
+# Use ExternalDNS Operator on Openshift in GCP environment
+
+### This document provides information about how to use the `ExternalDNS Operator` in OpenShift Container Platform on GCP.
+
+### Steps
+1. Export your cluster's kubeconfig
+```bash
+ $ export KUBECONFIG=/path/to/your/cluster/kubeconfig
+```
+
+2. Check user. The user shall have the access to `kube-system` namespace.
+
+```bash
+ $ oc whoami
+ system:admin
+```
+
+3. Copy the value of service_account.json in gcp-credentials secret in a file encoded-gcloud.json
+```bash
+$ oc get secret gcp-credentials -n kube-system --template='{{$v := index .data "service_account.json"}}{{$v}}' | base64 -d - > decoded-gcloud.json
+```
+
+4. Export Google credentials
+```bash
+$ export GOOGLE_CREDENTIALS=decoded-gcloud.json
+```
+
+5. Activate your account
+```bash
+gcloud auth activate-service-account  <client_email as per decoded-gcloud.json> --key-file=decoded-gcloud.json
+```
+
+6. Set your project
+```bash
+ gcloud config set project <project_id as per decoded-gcloud.json>
+```
+
+7. Get the routes to check the domain
+```bash
+oc get routes --all-namespaces | grep console
+openshift-console          console             console-openshift-console.apps.qe.gcp.devcluster.openshift.com                       console             https   reencrypt/Redirect     None
+openshift-console          downloads           downloads-openshift-console.apps.qe.gcp.devcluster.openshift.com                     downloads           http    edge/Redirect          None
+```
+
+8. Get your zone
+```bash
+$ gcloud dns managed-zones list | grep <dns name eg- As per the route the section after apps. i.e misalunk-azure.qe.azure.devcluster.openshift.com>
+```
+
+9. Check the zone doesn't have DNS records other than `NS` and `SOA`
+```bash
+$ gcloud dns record-sets list --zone=<the zone name you get from step 9>
+```
+
+10. Create a ExternalDNS CR
+```yaml
+$ cat <<EOF | oc create -f -
+apiVersion: externaldns.olm.openshift.io/v1alpha1
+kind: ExternalDNS
+metadata:
+  name: sample-gcp
+spec:
+  domains:
+    - filterType: Include
+      matchType: Exact
+      name: test-gcp1.qe.gcp.devcluster.openshift.com
+  provider:
+    type: GCP
+  source:
+    # Incase you have multiple ingress controllers you must specify ingress controller name in the routerName
+    # so that the external dns will use the router canonical name corresponding to it to create a dns record. 
+    openshiftRouteOptions:
+      routerName: default
+    #Source Type is route resource of OpenShift    
```

Comment:
```suggestion
    openshiftRouteOptions:
      routerName: default
```
Again to save some space. And similar to Azure doc, maybe a link to the different examples we have.